wandb/sdk/internal/handler.py

```diff
@@ -168,9 +169,24 @@ def _save_history(self, record: Record) -> None:
                 self._sampled_history[k].add(v)
 
     def handle_history(self, record: Record) -> None:
+        history_dict = proto_util.dict_from_proto_list(record.history.item)
+        # if syncing an old run, we can skip this logic
+        if history_dict.get("_step", None) is None:
+            has_step = record.history.HasField("step")
+            item = record.history.item.add()
+            item.key = "_step"
+            if has_step:
+                step = int(record.history.step.num)
+                history_dict["_step"] = step
+                item.value_json = json.dumps(step)
+                self._step = step + 1
+            else:
+                history_dict["_step"] = self._step
+                item.value_json = json.dumps(self._step)
+                self._step += 1
```

Comment:
Probably want to move this whole block into a function, something like:
```
def _history_assign_step(record):
```

Reply:
I went with def _history_assign_step, as part of the class, and it also took the history_dict as an argument. Thoughts?

Reply:
sounds good.